pinot-ingestion-jobs/pinot-hadoop/src/main/java/org/apache/pinot/hadoop/job/HadoopSegmentCreationJob.java

```diff
@@ -42,84 +38,31 @@
 import org.apache.hadoop.mapreduce.lib.output.TextOutputFormat;
 import org.apache.pinot.common.config.SegmentsValidationAndRetentionConfig;
 import org.apache.pinot.common.config.TableConfig;
-import org.apache.pinot.common.data.Schema;
 import org.apache.pinot.common.utils.StringUtil;
 import org.apache.pinot.hadoop.job.mappers.SegmentCreationMapper;
-import org.apache.pinot.hadoop.utils.JobPreparationHelper;
-import org.apache.pinot.hadoop.utils.PushLocation;
+import org.apache.pinot.hadoop.utils.PinotHadoopJobPreparationHelper;
+import org.apache.pinot.ingestion.common.JobConfigConstants;
+import org.apache.pinot.ingestion.jobs.SegmentCreationJob;
+import org.apache.pinot.ingestion.utils.JobPreparationHelper;
 
 
-public class SegmentCreationJob extends BaseSegmentJob {
-  protected static final String APPEND = "APPEND";
-
-  protected final Path _inputPattern;
-  protected final Path _outputDir;
-  protected final Path _stagingDir;
-  protected final String _rawTableName;
-
-  // Optional
-  protected final Path _depsJarDir;
-  protected final Path _schemaFile;
-  protected final String _defaultPermissionsMask;
-  protected final List<PushLocation> _pushLocations;
-
+public class HadoopSegmentCreationJob extends SegmentCreationJob {
   // Output Directory FileSystem
   protected FileSystem _outputDirFileSystem;
 
-  public SegmentCreationJob(Properties properties) {
+  public HadoopSegmentCreationJob(Properties properties) {
```

Comment:
Renaming this class is a backwards-incoompatible. Open source community can extend (and indeed, we do extend in Linkedin) this class to create their custom hadoop jobs.

Reply:
Ah wait, you have created a new SegmentCreationJob. Let me look over another time